### PR TITLE
 textarea minLength/maxLength

### PR DESCRIPTION
**Pull request by tigercosmos**
textarea minLength/maxLength

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21062)

### CODE REVIEW

**Comment by tigercosmos** [comment]
```
(view as text)
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: typeof IDL attribute", "test": "/html/dom/reflection-forms.html", "line": 93095, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: IDL get with DOM attribute unset", "test": "/html/dom/reflection-forms.html", "line": 93096, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to -2147483649", "test": "/html/dom/reflection-forms.html", "line": 93097, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to -2147483648", "test": "/html/dom/reflection-forms.html", "line": 93098, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to -36", "test": "/html/dom/reflection-forms.html", "line": 93099, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to -1", "test": "/html/dom/reflection-forms.html", "line": 93100, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to -0", "test": "/html/dom/reflection-forms.html", "line": 93101, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to 0", "test": "/html/dom/reflection-forms.html", "line": 93102, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to 1", "test": "/html/dom/reflection-forms.html", "line": 93103, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to 2147483647", "test": "/html/dom/reflection-forms.html", "line": 93104, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to 2147483648", "test": "/html/dom/reflection-forms.html", "line": 93105, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to 4294967295", "test": "/html/dom/reflection-forms.html", "line": 93106, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to 4294967296", "test": "/html/dom/reflection-forms.html", "line": 93107, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\"", "test": "/html/dom/reflection-forms.html", "line": 93108, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"-1\"", "test": "/html/dom/reflection-forms.html", "line": 93109, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"-0\"", "test": "/html/dom/reflection-forms.html", "line": 93110, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"0\"", "test": "/html/dom/reflection-forms.html", "line": 93111, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"1\"", "test": "/html/dom/reflection-forms.html", "line": 93112, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \" \\0\\x01\\x02\\x03\\x04\\x05\\x06\\x07 \\b\\t\\n\\v\\f\\r\\x0e\\x0f \\x10\\x11\\x12\\x13\\x14\\x15\\x16\\x17 \\x18\\x19\\x1a\\x1b\\x1c\\x1d\\x1e\\x1f  foo \"", "test": "/html/dom/reflection-forms.html", "line": 93113, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\\t7\"", "test": "/html/dom/reflection-forms.html", "line": 93114, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\\v7\"", "test": "/html/dom/reflection-forms.html", "line": 93115, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\\f7\"", "test": "/html/dom/reflection-forms.html", "line": 93116, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \" 7\"", "test": "/html/dom/reflection-forms.html", "line": 93117, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u00a07\"", "test": "/html/dom/reflection-forms.html", "line": 93118, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\ufeff7\"", "test": "/html/dom/reflection-forms.html", "line": 93119, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\\n7\"", "test": "/html/dom/reflection-forms.html", "line": 93120, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\\r7\"", "test": "/html/dom/reflection-forms.html", "line": 93121, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20287\"", "test": "/html/dom/reflection-forms.html", "line": 93122, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20297\"", "test": "/html/dom/reflection-forms.html", "line": 93123, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u16807\"", "test": "/html/dom/reflection-forms.html", "line": 93124, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u180e7\"", "test": "/html/dom/reflection-forms.html", "line": 93125, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20007\"", "test": "/html/dom/reflection-forms.html", "line": 93126, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20017\"", "test": "/html/dom/reflection-forms.html", "line": 93127, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20027\"", "test": "/html/dom/reflection-forms.html", "line": 93128, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20037\"", "test": "/html/dom/reflection-forms.html", "line": 93129, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20047\"", "test": "/html/dom/reflection-forms.html", "line": 93130, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20057\"", "test": "/html/dom/reflection-forms.html", "line": 93131, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20067\"", "test": "/html/dom/reflection-forms.html", "line": 93132, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20077\"", "test": "/html/dom/reflection-forms.html", "line": 93133, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20087\"", "test": "/html/dom/reflection-forms.html", "line": 93134, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u20097\"", "test": "/html/dom/reflection-forms.html", "line": 93135, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u200a7\"", "test": "/html/dom/reflection-forms.html", "line": 93136, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u202f7\"", "test": "/html/dom/reflection-forms.html", "line": 93137, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\u30007\"", "test": "/html/dom/reflection-forms.html", "line": 93138, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to undefined", "test": "/html/dom/reflection-forms.html", "line": 93139, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to 1.5", "test": "/html/dom/reflection-forms.html", "line": 93140, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to true", "test": "/html/dom/reflection-forms.html", "line": 93141, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to false", "test": "/html/dom/reflection-forms.html", "line": 93142, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to object \"[object Object]\"", "test": "/html/dom/reflection-forms.html", "line": 93143, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to NaN", "test": "/html/dom/reflection-forms.html", "line": 93144, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to Infinity", "test": "/html/dom/reflection-forms.html", "line": 93145, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to -Infinity", "test": "/html/dom/reflection-forms.html", "line": 93146, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to \"\\0\"", "test": "/html/dom/reflection-forms.html", "line": 93147, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to object \"2\"", "test": "/html/dom/reflection-forms.html", "line": 93148, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: setAttribute() to object \"3\"", "test": "/html/dom/reflection-forms.html", "line": 93149, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: IDL set to -2147483648", "test": "/html/dom/reflection-forms.html", "line": 93150, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: IDL set to -36", "test": "/html/dom/reflection-forms.html", "line": 93151, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: IDL set to -1", "test": "/html/dom/reflection-forms.html", "line": 93152, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: IDL set to 0", "test": "/html/dom/reflection-forms.html", "line": 93153, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: IDL set to 1", "test": "/html/dom/reflection-forms.html", "line": 93154, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.maxLength: IDL set to 2147483647", "test": "/html/dom/reflection-forms.html", "line": 93155, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: typeof IDL attribute", "test": "/html/dom/reflection-forms.html", "line": 93156, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: IDL get with DOM attribute unset", "test": "/html/dom/reflection-forms.html", "line": 93157, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to -2147483649", "test": "/html/dom/reflection-forms.html", "line": 93158, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to -2147483648", "test": "/html/dom/reflection-forms.html", "line": 93159, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to -36", "test": "/html/dom/reflection-forms.html", "line": 93160, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to -1", "test": "/html/dom/reflection-forms.html", "line": 93161, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to -0", "test": "/html/dom/reflection-forms.html", "line": 93162, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to 0", "test": "/html/dom/reflection-forms.html", "line": 93163, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to 1", "test": "/html/dom/reflection-forms.html", "line": 93164, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to 2147483647", "test": "/html/dom/reflection-forms.html", "line": 93165, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to 2147483648", "test": "/html/dom/reflection-forms.html", "line": 93166, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to 4294967295", "test": "/html/dom/reflection-forms.html", "line": 93167, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to 4294967296", "test": "/html/dom/reflection-forms.html", "line": 93168, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\"", "test": "/html/dom/reflection-forms.html", "line": 93169, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"-1\"", "test": "/html/dom/reflection-forms.html", "line": 93170, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"-0\"", "test": "/html/dom/reflection-forms.html", "line": 93171, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"0\"", "test": "/html/dom/reflection-forms.html", "line": 93172, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"1\"", "test": "/html/dom/reflection-forms.html", "line": 93173, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \" \\0\\x01\\x02\\x03\\x04\\x05\\x06\\x07 \\b\\t\\n\\v\\f\\r\\x0e\\x0f \\x10\\x11\\x12\\x13\\x14\\x15\\x16\\x17 \\x18\\x19\\x1a\\x1b\\x1c\\x1d\\x1e\\x1f  foo \"", "test": "/html/dom/reflection-forms.html", "line": 93174, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\\t7\"", "test": "/html/dom/reflection-forms.html", "line": 93175, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\\v7\"", "test": "/html/dom/reflection-forms.html", "line": 93176, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\\f7\"", "test": "/html/dom/reflection-forms.html", "line": 93177, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \" 7\"", "test": "/html/dom/reflection-forms.html", "line": 93178, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u00a07\"", "test": "/html/dom/reflection-forms.html", "line": 93179, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\ufeff7\"", "test": "/html/dom/reflection-forms.html", "line": 93180, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\\n7\"", "test": "/html/dom/reflection-forms.html", "line": 93181, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\\r7\"", "test": "/html/dom/reflection-forms.html", "line": 93182, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20287\"", "test": "/html/dom/reflection-forms.html", "line": 93183, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20297\"", "test": "/html/dom/reflection-forms.html", "line": 93184, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u16807\"", "test": "/html/dom/reflection-forms.html", "line": 93185, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u180e7\"", "test": "/html/dom/reflection-forms.html", "line": 93186, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20007\"", "test": "/html/dom/reflection-forms.html", "line": 93187, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20017\"", "test": "/html/dom/reflection-forms.html", "line": 93188, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20027\"", "test": "/html/dom/reflection-forms.html", "line": 93189, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20037\"", "test": "/html/dom/reflection-forms.html", "line": 93190, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20047\"", "test": "/html/dom/reflection-forms.html", "line": 93191, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20057\"", "test": "/html/dom/reflection-forms.html", "line": 93192, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20067\"", "test": "/html/dom/reflection-forms.html", "line": 93193, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20077\"", "test": "/html/dom/reflection-forms.html", "line": 93194, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20087\"", "test": "/html/dom/reflection-forms.html", "line": 93195, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u20097\"", "test": "/html/dom/reflection-forms.html", "line": 93196, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u200a7\"", "test": "/html/dom/reflection-forms.html", "line": 93197, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u202f7\"", "test": "/html/dom/reflection-forms.html", "line": 93198, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\u30007\"", "test": "/html/dom/reflection-forms.html", "line": 93199, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to undefined", "test": "/html/dom/reflection-forms.html", "line": 93200, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to 1.5", "test": "/html/dom/reflection-forms.html", "line": 93201, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to true", "test": "/html/dom/reflection-forms.html", "line": 93202, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to false", "test": "/html/dom/reflection-forms.html", "line": 93203, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to object \"[object Object]\"", "test": "/html/dom/reflection-forms.html", "line": 93204, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to NaN", "test": "/html/dom/reflection-forms.html", "line": 93205, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to Infinity", "test": "/html/dom/reflection-forms.html", "line": 93206, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to -Infinity", "test": "/html/dom/reflection-forms.html", "line": 93207, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to \"\\0\"", "test": "/html/dom/reflection-forms.html", "line": 93208, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to object \"2\"", "test": "/html/dom/reflection-forms.html", "line": 93209, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: setAttribute() to object \"3\"", "test": "/html/dom/reflection-forms.html", "line": 93210, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: IDL set to -2147483648", "test": "/html/dom/reflection-forms.html", "line": 93211, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: IDL set to -36", "test": "/html/dom/reflection-forms.html", "line": 93212, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: IDL set to -1", "test": "/html/dom/reflection-forms.html", "line": 93213, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: IDL set to 0", "test": "/html/dom/reflection-forms.html", "line": 93214, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: IDL set to 1", "test": "/html/dom/reflection-forms.html", "line": 93215, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "textarea.minLength: IDL set to 2147483647", "test": "/html/dom/reflection-forms.html", "line": 93216, "action": "test_result", "expected": "FAIL"}
```

File: components/script/dom/htmltextareaelement.rs [hunk 1]
```diff
@@ -110,6 +112,9 @@ static DEFAULT_COLS: u32 = 20;
 // https://html.spec.whatwg.org/multipage/#attr-textarea-rows-value
 static DEFAULT_ROWS: u32 = 2;
 
+static DEFAULT_MAX_LENGTH: i32 = -1;
+static DEFAULT_MIN_LENGTH: i32 = -1;
```
Comment (KiChjang) [inline components/script/dom/htmltextareaelement.rs]: I'm not sure why these two lines and DEFAULT_ROWS are statics, I believe they really should instead be const.

File: components/script/dom/htmltextareaelement.rs [hunk 2]
```diff
@@ -124,6 +129,8 @@ impl HTMLTextAreaElement {
             textinput: DomRefCell::new(TextInput::new(
                     Lines::Multiple, DOMString::new(), chan, None, None, SelectionDirection::None)),
             value_dirty: Cell::new(false),
+            maxlength: Cell::new(DEFAULT_MAX_LENGTH),
```
Comment (emilio) [inline components/script/dom/htmltextareaelement.rs]: I think these should be called `max_length` and `min_length` for consistency for the rest of the code.

File: tests/wpt/web-platform-tests/html/semantics/forms/the-textarea-element/textarea-maxlength.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+</head>
+
+<body>
+
+  <textarea id="none"></textarea>
+  <textarea id="negative" maxlength="-5"></textarea>
+  <textarea id="non-numeric" maxlength="not-a-number"></textarea>
+  <textarea id="assign-negative"></textarea>
```
Comment (emilio) [inline tests/wpt/web-platform-tests/html/semantics/forms/the-textarea-element/textarea-maxlength.html]: nit: You don't need these just to assign to it, right? You can use `document.createElement('textarea')` instead.

Comment (emilio) [reply tests/wpt/web-platform-tests/html/semantics/forms/the-textarea-element/textarea-maxlength.html]: We should probably also test that `setAttribute("maxlength")` changes the value reflected in `maxLength`, and vice versa.

File: components/script/dom/htmltextareaelement.rs (hunk 2 above)
Comment (emilio) [reply components/script/dom/htmltextareaelement.rs]: But these are actually unused, aren't they?

File: tests/wpt/web-platform-tests/html/semantics/forms/the-textarea-element/textarea-maxlength.html (hunk 3 above)
Comment (tigercosmos) [reply tests/wpt/web-platform-tests/html/semantics/forms/the-textarea-element/textarea-maxlength.html]: Actually these are textarea version modified from `html/semantics/forms/the-input-element/maxlength.html`.

About `setAttribute("maxlength")`, it has already tested well in ` tests/wpt/metadata/html/dom/reflection-forms.html`

Comment (tigercosmos) [reply tests/wpt/web-platform-tests/html/semantics/forms/the-textarea-element/textarea-maxlength.html]: Do you think we should refactor the code or this is ok?

Comment (emilio) [reply tests/wpt/web-platform-tests/html/semantics/forms/the-textarea-element/textarea-maxlength.html]: Yeah, I guess this is fine.

**Pull request merged**